src/middleware.ts

```diff
@@ -60,7 +60,11 @@ export function createMiddleware(
           res.statusMessage = statusText
 
           headers.forEach((value, name) => {
-            res.setHeader(name, value)
+            /**
+             * @note Use `.appendHeader()` to support multi-value
+             * response headers, like "Set-Cookie".
+             */
+            res.appendHeader(name, value)
```

Comment:
Previously, `res.setHeader()` would only set the last value of a multi-value cookie, like `Set-Cookie`.

Using `res.appendHeader()` will support both single- and multi-value response cookies.